src/pybind/mgr/dashboard/services/rbd.py

```diff
@@ -60,13 +60,29 @@ def format_features(features):
     return res
 
 
+def get_image_spec(pool_name, namespace, rbd_name):
+    namespace = '{}/'.format(namespace) if namespace else ''
+    return '{}/{}{}'.format(pool_name, namespace, rbd_name)
+
+
+def parse_image_spec(image_spec):
+    namespace_spec, image_name = image_spec.rsplit('/', 1)
+    if '/' in namespace_spec:
+        pool_name, namespace = namespace_spec.rsplit('/', 1)
+    else:
+        pool_name, namespace = namespace_spec, None
+    return pool_name, namespace, image_name
```

Comment:
The reason why I used this format was to follow the command-line `image_spec` format that is used to identify an image:

![Screenshot from 2019-10-28 12-16-53](https://user-images.githubusercontent.com/14297426/67677886-1a5c5e00-f97d-11e9-8a42-cabd5cad8591.png)

See https://docs.ceph.com/docs/master/man/8/rbd/

By using the command line `image_spec` I think we also guarantee that we don't break the API because any change to the `image_spec` format will be backward compatible.

The bigest challenge of changing URLs from `../{image_spec}/...` to `../pool/{p_name}/namespace/{n_name}/image/{i_name}..` is the fact that `namespace` is optional, so we will have to duplicate all endpoint, for instance:

Instead of having:
- `@ApiController('/block/image/{image_spec}/snap', ...)`

We will need both:
- `@ApiController('/block/pool/{p_name}/namespace/{n_name}/image/{i_name}/snap', ...)`
and 
- `@ApiController('/block/pool/{p_name}/image/{i_name}/snap', ...)`

Is this "duplication" a concern for you? Do you see a better way to avoid duplicating controllers due to the fact that `namespace` is optional?

Reply:
In this case, you usually keep the hierarchy but map the no-namespace to a namespace id that it's out of the expected range (e.g.: -1, none, ...).

Nevertheless, I'm seeing that it'll require a lot changes beyond the namespace part, so I just created this issue (https://tracker.ceph.com/issues/42514) to follow on this topic. Thanks anyway for your willingness!